awx/main/scheduler/kubernetes.py

```diff
@@ -51,6 +52,43 @@ def list_active_jobs(self, instance_group):
 
         return pods
 
+    @classmethod
+    def create_secret(self, instance_group, credential):
+        task = collections.namedtuple('Task', 'id instance_group')(id='', instance_group=instance_group)
+        pm = PodManager(task)
+        host = credential.get_input('host').split('/')[0]
+        username = credential.get_input("username")
+        password = credential.get_input("password")
+
+        # Construct container auth dict and base64 encode it
+        token = b64encode("{}:{}".format(username, password).encode('ascii')).decode()
+        auth_dict = json.dumps({"auths": {host: {"auth": token}}}, indent=4)
+        auth_data = b64encode(str(auth_dict).encode('ascii')).decode()
+
+        # Construct Secret object
+        secret = client.V1Secret()
+        secret.metadata = client.V1ObjectMeta(name="{}-image-pull-secret".format(task.instance_group.name.lower()))
+        secret.type = "kubernetes.io/dockerconfigjson"
+        secret.kind = "Secret"
+        secret.data = {".dockerconfigjson": auth_data}
+
+        # Create pullImageSecret in namespace, delete and create if it already exists
+        try:
+            pm.kube_api.create_namespaced_secret(namespace=pm.namespace, body=secret)
+        except client.rest.ApiException as e:
+            if e.status == 401:
```

Comment:
How likely is it to happen that the status isn't 401 or 409?  Should we have some kind of last-ditch catch here to handle it?